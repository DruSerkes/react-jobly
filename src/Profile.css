.Profile {
	padding-top: 5%;
}

.Profile form {
	background-color: white;
	width: 40vw;
	margin: 0 auto;
	border-radius: 10px;
	border: 1px solid lightgray;
	padding: 1rem;
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
}

.Profile h2 {
	font-size: 1.7rem;
	font-weight: 300;
}

.Profile h4 {
	font-size: 1.5rem;
	margin-bottom: 1rem;
}

.Profile p {
	margin: 0 0 2rem 0;
	font-size: 1.2rem;
}
