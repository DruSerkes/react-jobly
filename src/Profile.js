import React, { useContext } from 'react';
import { Formik, Form } from 'formik';
import TextInput from './forms/TextInput';
import userContext from './Context';
import JoblyApi from './JoblyApi';
import profileSchema from './forms/profileSchema';

const Profile = () => {
	const { currentUser, setCurrentUser } = useContext(userContext);

	const INITIAL_VALUES = {
		first_name : currentUser.first_name || '',
		last_name  : currentUser.last_name || '',
		email      : currentUser.email || '',
		photo_url  : currentUser.photo_url || '',
		password   : ''
	};

	const handleSubmit = async (values, { setSubmitting, resetForm }) => {
		for (let value in values) {
			if (!values[value]) delete values[value];
		}
		values.username = currentUser.username;
		setSubmitting(false);
		const user = await JoblyApi.updateUser(values);
		resetForm();
		setCurrentUser({ ...user });
	};

	return (
		<div className="Profile">
			<h2>Profile</h2>
			<Formik initialValues={INITIAL_VALUES} validationSchema={profileSchema} onSubmit={handleSubmit}>
				<Form>
					<h4>Username</h4>
					<p>{currentUser.username}</p>
					<TextInput label="First name" name="first_name" type="text" />
					<TextInput label="Last name" name="last_name" type="text" />
					<TextInput label="Email" name="email" type="email" />
					<TextInput label="Photo URL" name="photo_url" type="url" />
					<TextInput label="Re-enter Password" name="password" type="password" />
					<button type="submit">Save Changes</button>
				</Form>
			</Formik>
		</div>
	);
};

export default Profile;
