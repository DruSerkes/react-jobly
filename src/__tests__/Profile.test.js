import React from 'react';
import userContext from '../Context';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Profile from '../Profile';

describe('Profile tests', () => {
	it('renders without crashing', () => {
		let currentUser = {
			username   : 'test',
			first_name : 'test',
			last_name  : 'test',
			email      : 'test',
			logo_url   : 'test'
		};

		render(
			<userContext.Provider value={{ currentUser }}>
				<MemoryRouter>
					<Profile />
				</MemoryRouter>
			</userContext.Provider>
		);
	});

	it('matches snapshot', () => {
		let currentUser = {
			username   : 'test',
			first_name : 'test',
			last_name  : 'test',
			email      : 'test',
			logo_url   : 'test'
		};

		const { asFragment } = render(
			<userContext.Provider value={{ currentUser }}>
				<MemoryRouter>
					<Profile />
				</MemoryRouter>
			</userContext.Provider>
		);
		expect(asFragment()).toMatchSnapshot();
	});

	it('shows user Profile', () => {
		let currentUser = {
			username   : 'test',
			first_name : 'test',
			last_name  : 'test',
			email      : 'test',
			logo_url   : 'test'
		};
		const { getByText } = render(
			<userContext.Provider value={{ currentUser }}>
				<MemoryRouter>
					<Profile />
				</MemoryRouter>
			</userContext.Provider>
		);
		expect(getByText('Profile')).toBeInTheDocument();
	});
});
