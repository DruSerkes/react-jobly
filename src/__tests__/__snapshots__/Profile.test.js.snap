// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile tests matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Profile"
  >
    <h2>
      Profile
    </h2>
    <form
      action="#"
    >
      <h4>
        Username
      </h4>
      <p>
        test
      </p>
      <label
        for="first_name"
      >
        First name
      </label>
      <input
        class="text-input"
        name="first_name"
        type="text"
        value="test"
      />
      <label
        for="last_name"
      >
        Last name
      </label>
      <input
        class="text-input"
        name="last_name"
        type="text"
        value="test"
      />
      <label
        for="email"
      >
        Email
      </label>
      <input
        class="text-input"
        name="email"
        type="email"
        value="test"
      />
      <label
        for="photo_url"
      >
        Photo URL
      </label>
      <input
        class="text-input"
        name="photo_url"
        type="url"
        value=""
      />
      <label
        for="password"
      >
        Re-enter Password
      </label>
      <input
        class="text-input"
        name="password"
        type="password"
        value=""
      />
      <button
        type="submit"
      >
        Save Changes
      </button>
    </form>
  </div>
</DocumentFragment>
`;
